scripts/debug_tools/failure_lib.py

```diff
@@ -0,0 +1,47 @@
+import json
+import os
+
+
+def findPath(string, path_begin):
+    """ Find one path in string.
+    @return: the pair of the begin and end pos
+    """
+    path_end = string.find(' ', path_begin)
+    if path_end == -1:
+        path_end = len(string)
+    path = string[path_begin:path_end]
+    return path, path_end
+
+
+def changePaths(string, pathModifierFun):
+    """ Scan through the string and possibly replace all found paths.
+    @return: the modified string
+    """
+    result = ''
+    i = 0
+    while i < len(string):
```

Comment:
Using a for loop `for c in string` would not be easier? It is harder to make it wrong and create an infinite loop.

Reply:
Here we work with the positions and we jump forward in the range to skip a path, so the range based for cannot be used.